src/Server/src/GraphQL/Client/TokenRequestDataResultFactory.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using IdOps.Abstractions;
+using IdOps.Encryption;
+using IdOps.Model;
+using IdOps.Models;
+
+namespace IdOps.GraphQL;
+
+public class TokenRequestDataResultFactory : IResultFactory<TokenRequestData, TokenRequestInput>
+{
+    private readonly IEncryptionService _encryptionService;
+    private readonly IClientService _clientService;
+    private readonly IApiScopeService _scopeService;
+
+    public TokenRequestDataResultFactory(
+        IEncryptionService encryptionService,
+        IClientService clientService,
+        IApiScopeService apiScopeService)
+    {
+        _encryptionService = encryptionService;
+        _clientService = clientService;
+        _scopeService = apiScopeService;
+    }
+
+
+    public async Task<TokenRequestData> Create(TokenRequestInput input,
+        CancellationToken cancellationToken)
+    {
+        Client? client = await _clientService.GetByIdAsync(input.ClientId, cancellationToken);
+        if (client == null)
+        {
+            throw new ArgumentNullException($"Element with ID {input.ClientId} not found.");
+        }
+
+        var clientId = client.ClientId;
+
+        string secretEncrypted =
+            client.ClientSecrets.First(secret => secret.Id.Equals(input.SecretId)).EncryptedSecret;
+        var secretDecrypted =
+            await _encryptionService.DecryptAsync(secretEncrypted, cancellationToken);
+
+        var grantTypes = client.AllowedGrantTypes.First();
+
+        var scopeIds =
+            client.AllowedScopes.ToList().ConvertAll(clientScope => clientScope.Id );
```

Comment:
client.AllowedScopes.Select(x => x.Id).ToList() ?

Reply:
Maybe it doesnt even need to be a `ToList` when `scopeService` accepts a `IEnumerable<Guid>`